File: react/src/lib/domain/entities/BackupInterval.ts
import BaseEnum from './BaseEnum';

export default class BackupInterval extends BaseEnum {

    public static DAYS = new BackupInterval('1', 'project:backupsEnum:days');
    public static WEEKS = new BackupInterval('2', 'project:backupsEnum:weeks');
    public static MONTHS = new BackupInterval('3', 'project:backupsEnum:months');

    public constructor(id?: string, name?: string) {
        super(id, name);
    }
}

File: backend/src/main/java/com/equisoft/dca/backend/project/model/BackupInterval.java
package com.equisoft.dca.backend.project.model;

import com.equisoft.dca.backend.common.model.BaseEnum;

import lombok.Getter;

@Getter
public enum BackupInterval implements BaseEnum<Integer> {
	DAYS(1, "Days"),
	WEEKS(2, "Weeks"),
	MONTHS(3, "Months");

	private final int id;

	private final String name;

	BackupInterval(int id, String name) {
		this.id = id;
		this.name = name;
	}

	@Override
	public Integer getId() {
		return id;
	}
}
